test/stripe/stripe_object_test.rb

```diff
@@ -99,5 +99,98 @@ class StripeObjectTest < Test::Unit::TestCase
       # it to something more useful).
       assert_equal opts, source.instance_variable_get(:@opts)
     end
+
+    should "#serialize_params on an empty object" do
+      obj = Stripe::StripeObject.construct_from({})
+      assert_equal({}, Stripe::StripeObject.serialize_params(obj))
+    end
+
+    should "#serialize_params on a basic object" do
+      obj = Stripe::StripeObject.construct_from({ :foo => nil })
+      obj.update_attributes(:foo => "bar")
+      assert_equal({ :foo => "bar" }, Stripe::StripeObject.serialize_params(obj))
+    end
+
+    should "#serialize_params on a more complex object" do
+      obj = Stripe::StripeObject.construct_from({
+        :foo => Stripe::StripeObject.construct_from({
+          :bar => nil,
+          :baz => nil,
+        }),
+      })
+      obj.foo.bar = "newbar"
+      assert_equal({ :foo => { :bar => "newbar" } },
+        Stripe::StripeObject.serialize_params(obj))
+    end
+
+    should "#serialize_params on an array" do
+      obj = Stripe::StripeObject.construct_from({
+        :foo => nil,
+      })
+      obj.foo = ["new-value"]
+      assert_equal({ :foo => ["new-value"] },
+        Stripe::StripeObject.serialize_params(obj))
+    end
+
+    should "#serialize_params on an array that shortens" do
+      obj = Stripe::StripeObject.construct_from({
+        :foo => ["0-index", "1-index", "2-index"],
+      })
+      obj.foo = ["new-value"]
+      assert_equal({ :foo => ["new-value"] },
+        Stripe::StripeObject.serialize_params(obj))
+    end
+
+    should "#serialize_params on an array that lengthens" do
+      obj = Stripe::StripeObject.construct_from({
+        :foo => ["0-index", "1-index", "2-index"],
+      })
+      obj.foo = ["new-value"] * 4
+      assert_equal({ :foo => ["new-value"] * 4 },
+        Stripe::StripeObject.serialize_params(obj))
+    end
+
+    should "#serialize_params on an array of hashes" do
+      obj = Stripe::StripeObject.construct_from({
+        :foo => nil,
+      })
+      obj.foo = [
+        Stripe::StripeObject.construct_from({
+          :bar => nil
+        })
+      ]
+      obj.foo[0].bar = "baz"
+      assert_equal({ :foo => [{ :bar => "baz" }] },
+        Stripe::StripeObject.serialize_params(obj))
+    end
+
+    should "#serialize_params on an partially changed array of hashes" do
+      obj = Stripe::StripeObject.construct_from({
+        :foo => [
+          Stripe::StripeObject.construct_from({
+            :bar => nil
+          }),
+          Stripe::StripeObject.construct_from({
+            :bar => nil
+          }),
+        ]
+      })
+      obj.foo[1].bar = "baz"
+      assert_equal({ :foo => [{}, { :bar => "baz" }] },
```

Comment:
Do we want this to be the expected behavior? I thought we wanted to re-send the array every time.